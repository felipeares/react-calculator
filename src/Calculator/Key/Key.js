import React from "react";
import classes from "./Key.module.css";
import PropTypes from "prop-types";

const Key = props => (
  <div
    onClick={() => props.clicked(props.value)}
    className={[classes.Key, classes[props.keyType]].join(" ")}
  >
    <p className={classes.KeyValue}>{props.value}</p>
  </div>
);

Key.propTypes = {
  value: PropTypes.string.isRequired,
  clicked: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired
};

export default Key;
