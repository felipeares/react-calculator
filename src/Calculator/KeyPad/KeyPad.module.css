.KeyPad {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--keypad-height);
  padding: 2%;
  width: var(--keypad-width);
}

.NumbersContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: 80%;
  width: 75%;
}

.OperatorsContainer {
  display: flex;
  flex-direction: column;
  height: 80%;
  width: 25%;
}

.submit-container {
  height: 20%;
  width: 100%;
}
