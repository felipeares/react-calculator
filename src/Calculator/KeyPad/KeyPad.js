import React from "react";
import classes from "./KeyPad.module.css";
import PropTypes from "prop-types";
import Key from "../Key/Key";

const KeyPad = props => {
  const numberKeys = [
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9",
    ".",
    "0",
    "CE"
  ].map(number => {
    if (number === "CE")
      return (
        <Key
          key={number}
          value={number}
          clicked={props.resetPressed}
          keyType={"NumberKey"}
        />
      );
    return (
      <Key
        key={number}
        value={number}
        clicked={props.numberPressed}
        keyType={"NumberKey"}
      />
    );
  });

  const operatorKeys = ["+", "-", "*", "/"].map(operator => (
    <Key
      key={operator}
      value={operator}
      clicked={props.operatorPressed}
      keyType={"OperatorKey"}
    />
  ));

  return (
    <div className={classes.KeyPad}>
      <div className={classes.NumbersContainer}>{numberKeys}</div>
      <div className={classes.OperatorsContainer}>{operatorKeys}</div>
    </div>
  );
};

KeyPad.propTypes = {
  numberPressed: PropTypes.func.isRequired,
  dotPressed: PropTypes.func.isRequired,
  operatorPressed: PropTypes.func.isRequired,
  resetPressed: PropTypes.func.isRequired
};

export default KeyPad;
