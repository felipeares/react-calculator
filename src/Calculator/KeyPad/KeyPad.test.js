import React from "react";
import { shallow } from "enzyme";
import PropTypes from "prop-types";
import KeyPad from "./KeyPad";
import Key from "../Key/Key";

describe("KeyPad", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <KeyPad
        numberPressed={jest.fn()}
        dotPressed={jest.fn()}
        operatorPressed={jest.fn()}
        resetPressed={jest.fn()}
      />
    );
  });

  it("have not changed the UI", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 4 <div />'s", () => {
    expect(wrapper.find("div").length).toEqual(4);
  });

  it("should have a KeyPad named class on the first div", () => {
    expect(
      wrapper
        .find("div")
        .first()
        .hasClass("KeyPad")
    ).toEqual(true);
  });

  it("should check for all prop types correctly setup", () => {
    expect(KeyPad.propTypes.numberPressed).toBe(PropTypes.func.isRequired);
    expect(KeyPad.propTypes.dotPressed).toBe(PropTypes.func.isRequired);
    expect(KeyPad.propTypes.operatorPressed).toBe(PropTypes.func.isRequired);
    expect(KeyPad.propTypes.resetPressed).toBe(PropTypes.func.isRequired);
  });

  it("should render 12 <Key /> in the numbers container", () => {
    expect(wrapper.find(".NumbersContainer").find(Key).length).toEqual(12);
  });

  it("should render 4 <Key /> in the operators container", () => {
    expect(wrapper.find(".OperatorsContainer").find(Key).length).toEqual(4);
  });

  it("should render a Key within a SubmitContainer", () => {
    expect(wrapper.find(".SubmitContainer").find(Key).length).toEqual(1);
  });
});
