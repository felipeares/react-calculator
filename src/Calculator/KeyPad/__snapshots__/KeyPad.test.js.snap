// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad have not changed the UI 1`] = `
<div
  className="KeyPad"
>
  <div
    className="NumbersContainer"
  >
    <Key
      clicked={[MockFunction]}
      key="1"
      keyType="NumberKey"
      value="1"
    />
    <Key
      clicked={[MockFunction]}
      key="2"
      keyType="NumberKey"
      value="2"
    />
    <Key
      clicked={[MockFunction]}
      key="3"
      keyType="NumberKey"
      value="3"
    />
    <Key
      clicked={[MockFunction]}
      key="4"
      keyType="NumberKey"
      value="4"
    />
    <Key
      clicked={[MockFunction]}
      key="5"
      keyType="NumberKey"
      value="5"
    />
    <Key
      clicked={[MockFunction]}
      key="6"
      keyType="NumberKey"
      value="6"
    />
    <Key
      clicked={[MockFunction]}
      key="7"
      keyType="NumberKey"
      value="7"
    />
    <Key
      clicked={[MockFunction]}
      key="8"
      keyType="NumberKey"
      value="8"
    />
    <Key
      clicked={[MockFunction]}
      key="9"
      keyType="NumberKey"
      value="9"
    />
    <Key
      clicked={[MockFunction]}
      key="."
      keyType="NumberKey"
      value="."
    />
    <Key
      clicked={[MockFunction]}
      key="0"
      keyType="NumberKey"
      value="0"
    />
    <Key
      clicked={[MockFunction]}
      key="CE"
      keyType="NumberKey"
      value="CE"
    />
  </div>
  <div
    className="OperatorsContainer"
  >
    <Key
      clicked={[MockFunction]}
      key="+"
      keyType="OperatorKey"
      value="+"
    />
    <Key
      clicked={[MockFunction]}
      key="-"
      keyType="OperatorKey"
      value="-"
    />
    <Key
      clicked={[MockFunction]}
      key="*"
      keyType="OperatorKey"
      value="*"
    />
    <Key
      clicked={[MockFunction]}
      key="/"
      keyType="OperatorKey"
      value="/"
    />
  </div>
  <div
    className="SubmitContainer"
  >
    <Key
      clicked={[MockFunction]}
      keyType="SubmitKey"
      value="="
    />
  </div>
</div>
`;
